### DISCUSSION TITLE
where to start socket.io with redis ?

### DISCUSSION THREAD

**Discussion by magic-thomas** [socketio/socket.io#4802] [Q&A] [answered]
I am newbie to socket.io and redis.

I want to make socket app which use Redis Sub/Pub . 

I see two sections in the documentation.

https://socket.io/get-started/chat

https://socket.io/docs/v4/server-api/

Get-Start section use `require` but in Server Api section use `import`.  It is confusing.

Where should I start to learn socket.io? 

Can someone give me well documented and working examples for socket.io v4.x with Redis ?

**Comment by ruveydayilmaz** [comment, marked as answer]
Hey there!

The **Get Started** section is intended to offer you with a basic example to help you get started quickly. It includes the require syntax, which is widely used in Node.js modules.

The **Server API** section gets deeper into the Socket.IO and uses the import syntax, which is part of the ES6 module system.

You can use either approach. If you're unfamiliar with these technologies, I recommend beginning with the **Get Started** section, which provides a more straightforward introduction. 

I definitely suggest starting your learning journey with the official documentation. If you'd like to explore different examples, you can also check out these blog posts:
- https://dev.to/cglikpo/getting-started-with-socket-io-7m4
- https://socket.io/get-started/private-messaging-part-1
- https://blog.logrocket.com/building-real-time-location-app-node-js-socket-io

**Reply by magic-thomas** [reply]
Thank you. 

Your answer made me feel good about asking. 

Really thanks for your kindness.

**Reply by ruveydayilmaz** [reply]
You're always welcome :)
